src/Thermochimica-cxx.C

```diff
@@ -0,0 +1,333 @@
+  {
+    setUnitTemperature(tunit);
+    setUnitPressure(punit);
+    setUnitMass(munit);
+  }
+
+  void setStandardUnits()
+  {
+    TCAPI_setStandardUnits();
+  }
+
+  void setModelicaUnits()
+  {
+    TCAPI_setModelicaUnits();
+  }
+
+  void setTemperaturePressure(double temperature, double pressure)
+  {
+    TCAPI_setTemperaturePressure(&temperature, &pressure);
+  }
+
+  void setElementMass(int element, double mass)
+  {
+    TCAPI_setElementMass(&element, &mass);
+  }
+
+  int checkInfoThermo()
+  {
+    int idbg;
+    TCAPI_checkInfoThermo(&idbg);
+    return idbg;
+  }
+
+  void parseThermoFile()
+  {
+    TCAPI_sSParseCSDataFile();
+  }
+
+  void setPrintResultsMode(int mode)
+  {
+    TCAPI_setPrintResultsMode(&mode);
+  }
+
+  void printResults()
+  {
+    TCAPI_printResults();
+  }
+
+  void resetInfoThermo()
+  {
+    TCAPI_resetInfoThermo();
+  }
+
+  void resetThermo()
+  {
+    TCAPI_resetThermo();
+  }
+
+  void resetThermoAll()
+  {
+    TCAPI_resetThermoAll();
+  }
+
+  // re-initialization-related functions
+  void saveReinitData()
+  {
+    TCAPI_saveReinitData();
+  }
+
+  std::pair<int, int> getReinitDataSizes()
+  {
+    int elements, species;
+    TCAPI_getReinitDataSizes(&elements, &species);
+    return {elements, species};
+  }
+
+  std::vector<double> getMolesPhase()
+  {
+    auto [elements, species] = getReinitDataSizes();
+    std::vector<double> molesPhase(elements);
+    TCAPI_getMolesPhase(molesPhase.data());
+    return molesPhase;
+  }
+
+  std::vector<int> getAssemblage()
+  {
+    auto [elements, species] = getReinitDataSizes();
+    std::vector<int> assemblage(elements);
+    TCAPI_getAssemblage(assemblage.data());
+    return assemblage;
+  }
+
+  void setReinitRequested(bool requested)
+  {
+    int req = (requested) ? 1 : 0;
+    TCAPI_setReinitRequested(&req);
+  }
+
+  void resetReinit()
+  {
+    TCAPI_resetReinit();
+  }
+
+  std::vector<double> getAllElementPotential()
+  {
+    auto [elements, species] = getReinitDataSizes();
+    std::vector<double> potential(elements);
+    TCAPI_getAllElementPotential(potential.data());
+    return potential;
+  }
+
+  double getElementFraction(int atomicNumber)
+  {
+    double fraction;
+    TCAPI_getElementFraction(&atomicNumber, &fraction);
+
+    return fraction;
+  }
+
+  // Data extraction APIs
+
+  std::pair<double, int>
+  getOutputChemPot(const std::string &elementName)
+  {
+    double chemPot;
+    int info;
+    TCAPI_getOutputChemPot(elementName.c_str(), elementName.length(), &chemPot, &info);
+    return {chemPot, info};
+  }
+
+  std::tuple<double, double, int>
+  getOutputSolnSpecies(const std::string &phaseName, const std::string &speciesName)
+  {
+    double moleFrac, chemPot;
+    int info;
+    TCAPI_getOutputSolnSpecies(phaseName.c_str(), phaseName.length(), speciesName.c_str(), speciesName.length(), &moleFrac, &chemPot, &info);
+    return std::tuple<double, double, int>{moleFrac, chemPot, info};
```

Comment:
you should probably use `make_tuple` here instead:

```suggestion
    return std::make_tuple(moleFrac, chemPot, info);
```